File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        for i in range(len(arr)):
            curr = 0
            for j in range(i + 1, len(arr)):
                if curr < arr[j]:
                    curr = arr[j]
            arr[i] = curr
        arr[-1] = -1
        return arr

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length - 1;
        int maxL = 0;
        int maxR = 0;
        int max = 0;

        while (left <= right) {
            int curr = 0;

            if (maxL < heights[left])
                maxL = heights[left];

            if (maxR < heights[right])
                maxR = heights[right];

            if (maxL > maxR)
                max = Math.max(max, (maxR * (right - left)));
            else
                max = Math.max(max, (maxL * (right - left)));

            if (heights[left] < heights[right])
                left++;
            else
                right--;
        }

        return max;
    }
}
